sources/packages/api/src/services/student/student.service.ts

```diff
@@ -261,13 +261,13 @@ export class StudentService extends RecordDataModelService<Student> {
     lastName: string,
     appNumber: string,
   ): Promise<Student[]> {
-    const applicationExistsQuery = this.applicationRepo
-      .createQueryBuilder("application")
-      .where("application.applicationNumber Ilike :appNumber")
-      .andWhere("application.applicationStatus != :overwrittenStatus")
-      .select("1");
     const searchQuery = this.repo
       .createQueryBuilder("student")
+      .leftJoinAndSelect(
+        Application,
```

Comment:
In this query creating the proper join between students and applications?
What happens when we are searching only by the student names without providing an application number?
It seems that all the applications of the student will be part of the query, doesn't it?